src/openrct2/ride/Ride.cpp

```diff
@@ -3752,19 +3752,57 @@ static StationIndex ride_mode_check_valid_station_numbers(Ride* ride)
 
     switch (ride->mode)
     {
-        case RIDE_MODE_REVERSE_INCLINE_LAUNCHED_SHUTTLE:
-        case RIDE_MODE_POWERED_LAUNCH_PASSTROUGH:
-        case RIDE_MODE_POWERED_LAUNCH:
-        case RIDE_MODE_LIM_POWERED_LAUNCH:
+        case RideMode::ReverseInclineLaunchedShuttle:
+        case RideMode::PoweredLaunchPasstrough:
+        case RideMode::PoweredLaunch:
+        case RideMode::LimPoweredLaunch:
             if (numStations <= 1)
                 return 1;
             gGameCommandErrorText = STR_UNABLE_TO_OPERATE_WITH_MORE_THAN_ONE_STATION_IN_THIS_MODE;
             return 0;
-        case RIDE_MODE_SHUTTLE:
+        case RideMode::Shuttle:
             if (numStations >= 2)
                 return 1;
             gGameCommandErrorText = STR_UNABLE_TO_OPERATE_WITH_LESS_THAN_TWO_STATIONS_IN_THIS_MODE;
             return 0;
+        case RideMode::Normal:
+        case RideMode::ContinuousCircuit:
+        case RideMode::BoatHire:
+        case RideMode::UpwardLaunch:
+        case RideMode::RotatingLift:
+        case RideMode::StationToStation:
+        case RideMode::SingleRidePerAdmission:
+        case RideMode::UnlimitedRidesPerAdmission:
+        case RideMode::Maze:
+        case RideMode::Race:
+        case RideMode::Dodgems:
+        case RideMode::Swing:
+        case RideMode::ShopStall:
+        case RideMode::Rotation:
+        case RideMode::ForwardRotation:
+        case RideMode::BackwardRotation:
+        case RideMode::FilmAvengingAviators:
+        case RideMode::MouseTails3DFilm:
+        case RideMode::SpaceRings:
+        case RideMode::Beginners:
+        case RideMode::FilmThrillRiders:
+        case RideMode::StormChasers3DFilm:
+        case RideMode::SpaceRaiders3DFilm:
+        case RideMode::Intense:
+        case RideMode::Berserk:
+        case RideMode::HauntedHouse:
+        case RideMode::CircusShow:
+        case RideMode::DownwardLaunch:
+        case RideMode::CrookedHouse:
+        case RideMode::FreefallDrop:
+        case RideMode::ContinuousCircuitBlockSectioned:
+        case RideMode::PoweredLaunchBlockSectioned:
+        case RideMode::Count:
+        case RideMode::NullMode:
+        {
+            // This is workaround for multiple compilation errors of type "enumeration value ‘RIDE_MODE_*' not handled
+            // in switch [-Werror=switch]"
+        }
```

Comment:
I'd prefer a `default` case here instead.

Reply:
I actually suggested removing the `default`, because if we ever add another enumeration to the `RideMode´, compilation will fail instead of silently handling it.

Reply:
Without any of those (default or listing of all cases without action) there would be a lot of compilation warnings/errors regarding uncovered switch cases.

Reply:
@tupaschoal It's _extremely_ likely that a newly introduced ride mode would be treated the same here. I wouldn't suggest that applies to all places in the code, but I think it does here. Putting a case tower is not going to help with anything and makes for harder to read code.

Reply:
If you think new modes would probably fall into `default`, I'm ok with it, but I don't find this hard to read, specially with the comment.

Reply:
Yes, I'm quite sure 90% of new modes would.

@Lukasz-Pe Please change this to use `default`.